### ISSUE TITLE
[CI] SearchableSnapshotsIntegTests.testSnapshotOfSearchableSnapshotIncludesNoDataButCanBeRestored failing

### ISSUE THREAD

**Issue by mark-vieira** [elastic/elasticsearch#66411]
Looks like a newly added test that has failed several times now.

**Build scan**:
https://gradle-enterprise.elastic.co/s/tjveolnbh6o4s/tests/:x-pack:plugin:searchable-snapshots:internalClusterTest/org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests/testSnapshotOfSearchableSnapshotIncludesNoDataButCanBeRestored#1

**Repro line**:
`./gradlew ':x-pack:plugin:searchable-snapshots:internalClusterTest' --tests "org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests.testSnapshotOfSearchableSnapshotIncludesNoDataButCanBeRestored" -Dtests.seed=D4EDC366E82A039E -Dtests.security.manager=true -Dtests.locale=en-US -Dtests.timezone=Africa/Nouakchott -Druntime.java=8`

**Reproduces locally?**:
Nope.

**Applicable branches**:
So far only in `7.x`

**Failure history**:
https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=America/Los_Angeles&tests.container=org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests&tests.disabledDistributions=WyJvdXRjb21lOnNraXBwZWQiXQ&tests.sortField=FAILED&tests.test=testSnapshotOfSearchableSnapshotIncludesNoDataButCanBeRestored&tests.unstableOnly=true

**Failure excerpt**:
```


org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests > testSnapshotOfSearchableSnapshotIncludesNoDataButCanBeRestored FAILED
    SnapshotException[[qsbhbchqns:snapshot-2/x29NKSGUSHGk96DBGBtjkg] Indices don't have primary shards [xkhfnvpppc]]
        at __randomizedtesting.SeedInfo.seed([D4EDC366E82A039E:5BD22666E7678B24]:0)
        at org.elasticsearch.snapshots.SnapshotsService$2.execute(SnapshotsService.java:420)
        at org.elasticsearch.repositories.blobstore.BlobStoreRepository$1.execute(BlobStoreRepository.java:390)
        at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:59)
        at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:702)
        at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:324)
        at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:219)
        at org.elasticsearch.cluster.service.MasterService.access$000(MasterService.java:73)
        at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:151)
        at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150)
        at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188)
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:678)
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:252)
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:215)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

**Comment by mark-vieira** [comment]
@DaveCTurner this looks to be a new test you added. Mind taking a look?

**Comment by DaveCTurner** [comment]
Closed by #66415.
